akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpHeaderParser.scala

```diff
@@ -558,6 +558,7 @@ private[http] object HttpHeaderParser {
 
   @tailrec private def scanHeaderValue(hhp: HttpHeaderParser, input: ByteString, start: Int, limit: Int, log: LoggingAdapter,
                                        mode: IllegalResponseHeaderValueProcessingMode)(sb: JStringBuilder = null, ix: Int = start): (String, Int) = {
+    hhp.byteBuffer.clear()
```

Comment:
The byteBuffer will only be filled here and is also cleared in `hhp.decodeByteBuffer`. The bug was, that `decodeByteBuffer` might not be reached if `byteAt` fails with a `NotEnoughDataException`.